contracts/stakingv2/modules/StakeUnstake.sol

```diff
@@ -413,20 +424,21 @@ contract StakeUnstake is Common {
 
     function addAltRewards(address token, uint256 amount) public {
         if (amount != 0) {
-            _addAltRewards(token, amount);
+            (altRewardsPerShare[token], altRewardsPerSharePerBlock[token], altRewardsBlock[token])  = _addAltRewards(token, amount);
             IERC20(token).transferFrom(msg.sender, address(this), amount);
+            emit AddAltRewards(msg.sender, token, amount);
         }
     }
 
-    function _addAltRewards(address token, uint256 amount) internal {
+    function _addAltRewards(address token, uint256 amount) internal view
+        returns (uint256 _altRewardsPerShare, uint256 _altRewardsPerSharePerBlock, uint256 _altRewardsBlock){
         address poolAddress = token == SUSHI ? OOKI_ETH_LP : token;
-
         uint256 totalSupply = _totalSupplyPerToken[poolAddress];
         require(totalSupply != 0, "no deposits");
+        _altRewardsPerShare = altRewardsPerShare[token].add(amount.mul(1e12).div(totalSupply));
```

Comment:
Why does it need to be multiplied by 1e12 if the amount is in 8 decimals as well as totalSupply?

Reply:
totalSupply 18 decimals, to avoid getting 0